src/main/java/com/pusher/client/channel/impl/ChannelManager.java

```diff
@@ -137,6 +152,12 @@ public void run() {
 
                 if (connection.getState() == ConnectionState.CONNECTED) {
                     try {
+
+                        // only private encrypted channels can prepare themselves atm
+                        if (channel instanceof PrivateEncryptedChannelImpl) {
```

Comment:
this change is nested inside a `factory.queueOnEventThread(new Runnable() {` which is necessary to actually have access to the authorizer data at this point.